src/week-view.c

```diff
@@ -196,44 +211,91 @@ static void week_view_draw(WeekView* wv, cairo_t* cr)
 	}
 	g_object_unref(layout);
 
-	// draw vertical lines for days
+	// vertical lines for days
 	cairo_set_dash(cr, NULL, 0, 0);
+
 	icaltimetype day = icaltime_from_timet_with_zone(wv->current_view.start, 1, wv->current_tz);
 	for (int d = 0; d < num_days; ++d, icaltime_adjust(&day, 1, 0, 0, 0)) {
 		double x = wv->x + SIDEBAR_WIDTH + d * day_width;
-		char daylabel[16];
+		char day_label[16];
 		time_t tt = icaltime_as_timet(day);
 		struct tm* t = localtime(&tt);
 
-		strftime(daylabel, 16, "%e", t);
-		cairo_move_to(cr, x + 5, wv->y + HEADER_HEIGHT - 5);
-		cairo_set_font_size(cr, 22);
-		cairo_set_source_rgb(cr, dark, dark, dark);
-		cairo_show_text(cr, daylabel);
+		// day of month
+		strftime(day_label, 16, "%e", t);
+		cairo_move_to(cr, x + 8, wv->y + HEADER_HEIGHT - 14);
+		cairo_set_font_size(cr, 14);
+
+		if (day.month == wv->now.month && day.day == wv->now.day) {
+			gdk_cairo_set_source_rgba(cr, color_current_day);
+		} else {
+			gdk_cairo_set_source_rgba(cr, color_fg);
+		}
+		cairo_show_text(cr, day_label);
 
-		strftime(daylabel, 16, "%a", t);
+		// weekday abbreviation
+		strftime(day_label, 16, "%a", t);
 
-		for (char* p = daylabel; *p; ++p)
+		for (char* p = day_label; *p; ++p)
 			*p = toupper(*p);
-		cairo_move_to(cr, x + 35, wv->y + HEADER_HEIGHT - 5);
-		cairo_set_font_size(cr, 11);
-		cairo_show_text(cr, daylabel);
+		cairo_move_to(cr, x + 30, wv->y + HEADER_HEIGHT - 14);
+		cairo_set_font_size(cr, 14);
+		cairo_show_text(cr, day_label);
 
-		cairo_set_source_rgb(cr, med, med, med);
+		gdk_cairo_set_source_rgba(cr, color_fg_50);
 		cairo_move_to(cr, x, wv->y + HEADER_HEIGHT);
 		cairo_rel_line_to(cr, 0, wv->height);
 		cairo_stroke(cr);
+
+		gdk_cairo_set_source_rgba(cr, color_header_divider);
+		cairo_move_to(cr, x, wv->y);
+		cairo_rel_line_to(cr, 0, HEADER_HEIGHT);
+		cairo_stroke(cr);
 	}
 
 	// top bar
-	cairo_set_source_rgb(cr, dark, dark, dark);
+	gdk_cairo_set_source_rgba(cr, color_bg);
 	cairo_move_to(cr, wv->x, wv->y + HEADER_HEIGHT);
 	cairo_rel_line_to(cr, wv->width, 0);
 	cairo_move_to(cr, wv->x, wv->y + day_begin_yoffset);
 	cairo_rel_line_to(cr, wv->width, 0);
 	cairo_stroke(cr);
 }
 
+void allocate_colors(void) {
+	color_bg = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_header_bg = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_header_divider = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_fg = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_fg_50 = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_current_time_marker = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_current_day = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+}
+
+void init_style(GtkWidget *mainWindow) {
```

Comment:
I don't like the way `mainWindow` gets involved here. Typically, the best place for this kind of function would be `week_view_class_init`, which gets called only once no matter how many `WeekView`s are created. `WeekView` is also a `GtkWidget` so you should be able to call `gtk_widget_get_style_context` on it.

I do remember reading somewhere that the style is not available until the widget is realized. This is presumably to support different instances having different colors. In this case, I would suggest making two static structures `colors_dark` and `colors_light`, connecting a handler to `WeekView`s `realized` signal, and updating a new pointer in the `WeekView` instance to point either to `colors_dark` or `colors_light` depending on the style context.

Reply:
i did not find a way to get a WeekView reference that would allow retrieval of the style context within week_view_class_init, therefor moved it into week_view_init first of all.
I suggest using a programmatic initialization instead of static structures, as it allows for a mix of 1. retrievable colors, plus 2. generical calculation of inbetween colors which are missing in style context (unless using alpha values is sufficient for those), and 3. static color values for dark and light colors.

Reply:
Right, you cannot get a `WeekView` instance from `week_view_class_init`. What I meant was something more like this:
```
struct WeekViewColors {
  GdkRGBA bg;
  ...
};

static struct WeekViewColors colors_light, colors_dark;

struct _WeekView {
  ...
  WeekViewColors* colors;
  ...
};

...

static void week_view_class_init(WeekViewClass* klass)
{
  ...
  gdk_rgba_parse_color(colors_light.bg, "#fff");
  gdk_rgba_parse_color(colors_dark.bg, "#000");
  ...
}

static void on_realize(WeekView* wv)
{
  ...
  // retrieve gtkstylecontext
  if(is_dark_from_gtkstylecontext)
    wv->colors = &colors_dark;
  else
    wv->colors = &colors_light;
  ...
}
```

This way, the colors are only allocated once even if there are multiple `WeekView` instances, and each instance does the style check and updates its attribute to point to the appropriate static struct

Reply:
Actually I think it's also OK to allocate the colours directly on discovering the `GtkStyleContext` theme type, as you currently do. In practice we won't have multiple `WeekView` instances, but if we did it would be nice that they wouldn't all contain a duplicate struct of colours, but just a pointer to a shared struct. So you could consider this a micro-optimisation. Anyway probably best to just prefer whichever approach you find the most readable